notion-core/src/main/scala/com/ambiata/notion/core/LocationIO.scala

```diff
@@ -76,6 +76,27 @@ case class LocationIO(configuration: Configuration, s3Client: AmazonS3Client) {
         Hdfs.readWith(new Path(path), f).run(configuration)
     }
 
+  /** write to a given location, using an OutputStream */
+  def writeUnsafe(location: Location)(f: java.io.OutputStream => RIO[Unit]): RIO[Unit] =
+    location match {
+      case l @ LocalLocation(_) =>
+        RIO.safe(l.dirPath.dirname.toFile.mkdirs) >>
+        RIO.using(l.filePath.toOutputStream)(f)
+
+      case HdfsLocation(path) =>
+        Hdfs.writeWith(new Path(path), f).run(configuration)
+
+      case S3Location(bucket, key) =>
+        for {
+          tmpFile <- RIO.safe(Temporary.uniqueFilePath)
+          out     <- RIO.safe(new java.io.FileOutputStream(tmpFile.toFile))
```

Comment:
This could just be cleaned up a bit, these are already exposed
`FilePath#toOutputStream`
`FilePath#toInputStream`